### PR TITLE
Implement instance permissions/role

### LINKED ISSUE #773: Support Instance roles - permissions
> Some applications rely on a "trusted agent" (to steal aws' terminology) system to determine their access to other systems.

Allow the specification of this role/service-account for the instance to be created as.
- AWS `instance-profile`
- GCP `service account email`
- Azure `system-assigned managed identity` ? (I haven't used enough of azure)

------
Some quick research shows this to be supported by all of the cloud providers supported so far
### AWS
https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_use_switch-role-ec2.html
https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html

### GCP
https://cloud.google.com/compute/docs/access/create-enable-service-accounts-for-instances

### Azure
https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/qs-configure-portal-windows-vm?context=/azure/virtual-machines/context/context

### PR DESCRIPTION
**Pull request by dacbd**
- Closes #773 

I can close the draft if preferred, but I figured I'd put it up for the sake of visibility.

⚠️ **Requires upstream changes**
- will rebase out testing changes

### CODE REVIEW

File: src/terraform.js [hunk 1]
```diff
@@ -56,7 +56,7 @@ const iterativeProviderTpl = () => {
 terraform {
   required_providers {
     iterative = {
-      source = "iterative/iterative"
+      source = "github.com/iterative/iterative"
```
Comment (DavidGOrtega) [inline src/terraform.js:59]: @dacbd is this intended or maybe you forgot to change it back maybe?

**Review by DavidGOrtega — CHANGES_REQUESTED** [review]

File: src/terraform.js (hunk 1 above)
Comment (dacbd) [reply src/terraform.js:59]: Intended, I'll rebase the first commit out or change it back before it's done/marked as ready for review.

**Comment by dacbd** [comment]
https://github.com/iterative/terraform-provider-iterative/pull/234

**Comment by DavidGOrtega** [comment]
@dacbd TPI v0.8.0 is released 🚀 
Please update the PR when you are able to 🙏

**Comment by dacbd** [comment]
> @dacbd TPI v0.8.0 is released 🚀 Please update the PR when you are able to 🙏

❤️  Thanks! testing a couple of things out right now. Try to check a proxy issue I ran into.

**Comment by dacbd** [comment]
Thanks Much! @DavidGOrtega, going to test out a couple more things but it looks to work as expected so far!

File: bin/cml/runner.js [hunk 2]
```diff
@@ -470,6 +473,12 @@ exports.builder = (yargs) =>
         description:
           'Instance type. Choices: [m, l, xl]. Also supports native types like i.e. t2.micro'
       },
+      cloudPermissionSet: {
+        type: 'string',
+        default: '',
+        description:
```
Comment (dacbd) [inline bin/cml/runner.js:479]: @DavidGOrtega should this be hidden?

**Comment by dacbd** [comment]
Add test to terraform.test.js?

File: bin/cml/runner.js (hunk 2 above)
Comment (DavidGOrtega) [reply bin/cml/runner.js:479]: @dacbd sorry, what do you mean by hidden?

**Comment by DavidGOrtega** [comment]
If you can! the test is easy just modify the input in current test and regenerate the template

```sh
npm run do_snapshots
```

**Comment by DavidGOrtega** [comment]
I can do this for you if you want me to do it. I would have done it but Im not sure if you want to do everything on your own.

File: bin/cml/runner.js (hunk 2 above)
Comment (dacbd) [reply bin/cml/runner.js:479]: Yargs has a hidden attr you can use

**Comment by dacbd** [comment]
> I can do this for you if you want me to do it. I would have done it but Im not sure if you want to do everything on your own.

I would like to hand off something that is as complete as possible 🚀

File: bin/cml/runner.js (hunk 2 above)
Comment (DavidGOrtega) [reply bin/cml/runner.js:479]: Ah! I see. I do not see any reason to hide it. Indeed is a new interesting security feature

**Comment by DavidGOrtega** [comment]
> I would like to hand off something that is as complete as possible 🚀

👍 
go for it! 🚀

**Review by DavidGOrtega — APPROVED** [review]
🚀 

Thanks a lot for the effort @dacbd

**Pull request merged**